test/illuminate/Safe.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: UNLICENSED
+// Adapted from: https://github.com/Rari-Capital/solmate/blob/main/src/utils/SafeTransferLib.sol
+
+pragma solidity 0.8.13;
+
+import {IErc20} from "./Interfaces.sol";
+/**
+  @notice Safe ETH and ERC20 transfer library that gracefully handles missing return values.
+  @author Modified from Gnosis (https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2SafeERC20.sol)
+  @dev Use with caution! Some functions in this library knowingly create dirty bits at the destination of the free memory pointer.
+*/
+
+library Safe {
+  /// @param e Erc20 token to execute the call with
+  /// @param t To address
+  /// @param a Amount being transferred
+  function approve(IErc20 e, address t, uint256 a) internal {
+    bool result;
+
+    assembly {
+      // Get a pointer to some free memory.
+      let pointer := mload(0x40)
+
+      // Write the abi-encoded calldata to memory piece by piece:
+      mstore(pointer, 0x095ea7b300000000000000000000000000000000000000000000000000000000) // Begin with the function selector.
+      mstore(add(pointer, 4), and(t, 0xffffffffffffffffffffffffffffffffffffffff)) // Mask and append the "to" argument.
+      mstore(add(pointer, 36), a) // Finally append the "amount" argument. No mask as it's a full 32 byte value.
+
+      // Call the token and store if it succeeded or not.
+      // We use 68 because the calldata length is 4 + 32 * 2.
+      result := call(gas(), e, 0, pointer, 68, 0, 0)
+    }
+
+    require(success(result), "approve failed");
+  }
+
+  /// @param e Erc20 token to execute the call with
```

Comment:
Several methods commented out - was this intentional?

Reply:
yes. it is unknown exactly what methods this will need. uncomment them as needed, then when the contract is done - delete the ones never used.